examples/plot_vgm_error.py

```diff
@@ -0,0 +1,256 @@
+cbar = plt.colorbar()
+cbar.set_label('Elevation differences (m)')
+plt.show()
+
+
+# %%
+# We clearly see that the residual elevation differences on stable terrain are not random. The positive and negative
+# differences (blue and red, respectively) appear correlated over large distances.
+#
+# Conclusion: **These correlated errors need to be quantified to reliably estimate elevation measurement errors for a
+# sum, or average of elevation differences samples for a specific surface area**.
+
+# %%
+# Additionally, we notice that the elevation differences are still polluted by unrealistically large elevation
+# differences near glaciers, probably because the glacier inventory is more recent than the data, and the outlines are too small.
+# To remedy this, we filter large elevation differences outside 4 NMAD.
+dh.data[np.abs(dh.data) > 4 * xdem.spatialstats.nmad(dh.data)] = np.nan
+
+# %%
+# We plot the elevation differences after filtering to check that we successively removed the reminaing glacier signals.
+plt.figure(figsize=(8, 5))
+plt_extent = [
+    dh.bounds.left,
+    dh.bounds.right,
+    dh.bounds.bottom,
+    dh.bounds.top,
+]
+plt.imshow(dh.data.squeeze(), cmap="RdYlBu", vmin=-4, vmax=4, extent=plt_extent)
+cbar = plt.colorbar()
+cbar.set_label('Elevation differences (m)')
+plt.show()
+
+# %%
+# To quantify the spatial correlation of the data, we sample an empirical variogram.
+# The empirical variogram describes the variance between the elevation differences of pairs of pixels depending on their
+# distance. This distance between pairs of pixels if referred to as spatial lag.
+# To perform this procedure effectively, we use improved methods that provide efficient pairwise sampling methods for
+# large grid data in `scikit-gstat <https://mmaelicke.github.io/scikit-gstat/index.html>`_, which are encapsulated
+# conveniently by :func:`xdem.spatialstats.sample_multirange_variogram`:
+df = xdem.spatialstats.sample_multirange_variogram(
+    values=dh.data, gsd=dh.res[0], subsample=50, runs=30, nrun=10, estimator='cressie')
+
+# %%
+# We plot the empirical variogram:
+xdem.spatialstats.plot_vgm(df)
+
+# %%
+# With this plot, it is hard to conclude anything.
+# Properly visualizing the empirical variogram is one of the most important step. With grid data, we expect short-range
+# correlations close to the resolution of the grid (~20-200 meters), but also possibly longer range correlation due to
+# instrument noise or alignment issues (~1-50 km) (Hugonnet et al., in prep).
+# To better visualize the variogram, we can either change the axis to log-scale, but this might make it more difficult
+# to later compare to variogram models.
+# Another solution is to split the variogram plot into subpanels, each with its own linear scale:
+
+# %%
+# **Log scale:**
+xdem.spatialstats.plot_vgm(df, xscale='log')
+
+# %%
+# **Subpanels with linear scale:**
+xdem.spatialstats.plot_vgm(df, xscale_range_split=[100, 1000, 10000])
```

Comment:
Fantastic tools! Make variograms seem so easy !!

Reply:
=D
The new sampling makes all the difference in terms of speed, and quality of results :+1: